Code/Framework/AzFramework/AzFramework/Components/NativeUISystemComponent.cpp

```diff
@@ -0,0 +1,146 @@
+        }
+
+        if ((m_nativeWindowImplFactory = GetNativeWindowImplementationFactory()) != nullptr)
+        {
+            AZ::Interface<NativeWindow::ImplementationFactory>::Register(m_nativeWindowImplFactory);
+        }
+    }
+
+    NativeUISystemComponent::~NativeUISystemComponent()
+    {
+        if (m_applicationImplFactory != nullptr)
+        {
+            AZ::Interface<Application::ImplementationFactory>::Unregister(m_applicationImplFactory);
+            delete m_applicationImplFactory;
+        }
+
+        if (m_deviceGamepadImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceGamepad::ImplementationFactory>::Unregister(m_deviceGamepadImplFactory);
+            delete m_deviceGamepadImplFactory;
+        }
+        
+        if (m_deviceKeyboardImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceKeyboard::ImplementationFactory>::Unregister(m_deviceKeyboardImplFactory);
+            delete m_deviceKeyboardImplFactory;
+        }
+
+        if (m_deviceMotionImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceMotion::ImplementationFactory>::Unregister(m_deviceMotionImplFactory);
+            delete m_deviceMotionImplFactory;
+        }
+
+        if (m_deviceMouseImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceMouse::ImplementationFactory>::Unregister(m_deviceMouseImplFactory);
+            delete m_deviceMouseImplFactory;
+        }
+
+        if (m_deviceTouchImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceTouch::ImplementationFactory>::Unregister(m_deviceTouchImplFactory);
+            delete m_deviceTouchImplFactory;
+        }
+
+        if (m_deviceVirtualKeyboardImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceVirtualKeyboard::ImplementationFactory>::Unregister(m_deviceVirtualKeyboardImplFactory);
+            delete m_deviceVirtualKeyboardImplFactory;
+        }
+
+        if (m_nativeWindowImplFactory != nullptr)
+        {
+            AZ::Interface<NativeWindow::ImplementationFactory>::Unregister(m_nativeWindowImplFactory);
+            delete m_deviceVirtualKeyboardImplFactory;
+        }
+    }
+
+    void NativeUISystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<AzFramework::NativeUISystemComponent, AZ::Component>()
+                ->Version(1)
+                ;
+        }
+    }
+
+    void NativeUISystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+    }
+
+    void NativeUISystemComponent::GetIncompatibleServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC("NativeUIInputSystemService", 0x67675d29));
```

Comment:
This should use `AZ_CRC_CE` which calculates the CRC at compile time and is not subject to user error around keeping the string and the integer value in sync.
```suggestion
        incompatible.push_back(AZ_CRC_CE("NativeUIInputSystemService"));
```